### ISSUE TITLE
Add pause / resume just like flowder

### ISSUE THREAD

**Issue by prateekmedia** [dart-tools/dio_http#6]
<details>
<summary><b>New Issue Checklist</b></summary>

- [x] I have searched for a similar issue in the [project](https://github.com/dart-tools/dio_http/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | flutter                          |      |
| Platform Version               | any                              |      |
| Dio Version                    | latest                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

</details>

### Issue Description and Steps

There were many stale issues on dio's issue tracker like flutterchina/dio#1020 flutterchina/dio#399 flutterchina/dio#395. Pause and resume for downloads are always good for a better downloading experience.

This can be added by replacing `CancelToken` with more advanced `DownloadToken` or `DioToken`, which will contain all action like .cancel, .pause, .resume or isCancelled, isPaused, isResumed or something like that.

This can be inspired from an existing package called [`flowder`](https://pub.dev/packages/flowder).

At last I wish `dio_http` will became a successor of `dio` and will contribute back to `dio` when it get's maintainance back on track.

**Comment by jgoyvaerts** [comment]
Both this issue and the issue regarding multi threaded downloading (https://github.com/dart-tools/dio_http/issues/3) seem very focused on using dio_http as a file downloading client, which is not how most people are using dio_http (most people use it for managing their api requests, which are usually simple json GET/POST requests).

I'm not sure if adding both of those features would fit better in dio_http, or in a separate "file downloader" package. Is there a reason why you want this functionality in dio_http instead of using flowder like you mentioned?

**Comment by prateekmedia** [comment]
Firstly `flowder` is not actively maintained project and also doesn't have multi-threading either. Also it uses `dio` under the hood and that means that these function can be ported to this library too.

Sure many people use this as a http client and not as a download manager, but some people(like me) who like to have multi-platform downloading support uses `dio`/`dio-http` instead in their project's.

Also `dio`/`dio-http` is advertised as a download client too in the README.

"A powerful Http client for Dart, which supports ..., **File downloading**, ... etc."

At last it is upto the maintainers / contributors to add it and it will probably help the project in the long run.

**Comment by jgoyvaerts** [comment]
This package is now archived. This was originally forked because the owner of the [Dio](https://github.com/flutterchina/dio) package was inactive, but he has sinced returned from inactiveness and has added some more contributors to prevent this from happening again in the future. All future development continues over on [Dio](https://github.com/flutterchina/dio).
